test/hummingbot/connector/exchange/bitmart/test_bitmart_api_order_book_data_source.py

```diff
@@ -344,21 +351,28 @@ def _trade_ws_messsage(self):
         }
         return ujson.dumps(resp)
 
-    @patch('websockets.connect', new_callable=AsyncMock)
+    @patch('aiohttp.ClientSession.ws_connect', new_callable=AsyncMock)
     def test_listen_for_trades(self, mock_ws):
-        msg_queue: asyncio.Queue = asyncio.Queue()
+        msg_queue: asyncio.Queue = BitmartAPIOrderBookDataSourceUnitTests.QueueMock()
+
         mock_ws.return_value = self.mocking_assistant.create_websocket_mock()
         mock_ws.close.return_value = None
 
         # Add message to be processed after subscriptions, to unlock the test
-        self.mocking_assistant.add_websocket_text_message(mock_ws.return_value, self._trade_ws_messsage())
+        self.mocking_assistant.add_websocket_aiohttp_message(mock_ws.return_value, bitmart_utils.compress_ws_message(self._trade_ws_messsage()), message_type=aiohttp.WSMsgType.BINARY)
         BitmartAPIOrderBookDataSource._trading_pairs = ["ETH-USDT"]
 
         self.listening_task = self.ev_loop.create_task(
             self.data_source.listen_for_trades(self.ev_loop, msg_queue)
         )
-        trade1: OrderBookMessage = self.ev_loop.run_until_complete(msg_queue.get())
-        trade2: OrderBookMessage = self.ev_loop.run_until_complete(msg_queue.get())
+
+        try:
+            self.async_run_with_timeout(msg_queue.get())
+        except asyncio.exceptions.TimeoutError:
+            pass
+
+        trade1: OrderBookMessage = msg_queue.messages[-2]
+        trade2: OrderBookMessage = msg_queue.messages[-1]
```

Comment:
Same idea as above, but it can be done twice:
```
trade1 = self.async_run_with_timeout(msg_queue.get())
trade2 = self.async_run_with_timeout(msg_queue.get())
```